### ISSUE TITLE
ComBat_seq took a very long time

### ISSUE THREAD

**Issue by rongwong1119** [zhangyuqing/ComBat-seq#29]
Hi, I was using ComBat_seq for RNA seq data, and I have covar_mod as the argument, which has 8 variables, however, it has been run more than 10 hours, and still not completed, which is wired.

I wonder why it took so long?

**Comment by DamienTan** [comment]
I have the same issue, have you solved this problem? @rongwong1119

**Comment by pyunhyojin5504** [comment]
I have the same issue too, in fact I only have 2 covariates but still for over 15 hours, it is in process. It would be really helpful if you share how long it took..! I'm wondering if I have to execute it or not.. Thanks!  @rongwong1119 @DamienTan

**Comment by rongwong1119** [comment]
I now used two covariates, and it took only a few minutes.

**Comment by rongwong1119** [comment]
I have two covariates, and it took a few minutes.

**Comment by pyunhyojin5504** [comment]
@rongwong1119  Thanks for your quick reply! I'm sure that somethings wrong with my job then... One thing more just in case, I have 1300 samples and 200 genes.. is this matrix size pretty similar to your work? If it is even smaller than your work, I should really abort and check other solutions...!! Thanks again for your quick reply :)

**Comment by DamienTan** [comment]
> I have the same issue too, in fact I only have 2 covariates but still for over 15 hours, it is in process. It would be really helpful if you share how long it took..! I'm wondering if I have to execute it or not.. Thanks! @rongwong1119 @DamienTan

Yep, I used the argument "covar_mod" to add 2 covariates, but it took me more than 4 days and was still running. Finally, I killed this process. Maybe only using the argument "group" can avoid this problem, but there is another problem, which will create some very large values at the adjusted matrix. #20

**Comment by rongwong1119** [comment]
I have 20000+ genes and 60 samples

**Comment by maxnest** [comment]
Hi all! I faced a similar problem: I have 2 covariates in 'covar_mod' matrix and the analysis has been running for 7 days already. @DamienTan did you manage to solve this problem?

**Comment by DamienTan** [comment]
> 

Sorry, I didn't solve it!  I gave up. @maxnest

**Comment by maxnest** [comment]
> 
@DamienTan 
Thanks for the answer. Have you used any other program?

**Comment by DamienTan** [comment]
> > 
> 
> @DamienTan Thanks for the answer. Have you used any other program?

@maxnest Haven't! as I commented above, there were another problem using the argument "group". But I didn't use other program.

**Comment by rohanr3ji** [comment]
> > I have the same issue too, in fact I only have 2 covariates but still for over 15 hours, it is in process. It would be really helpful if you share how long it took..! I'm wondering if I have to execute it or not.. Thanks! [@rongwong1119](https://github.com/rongwong1119) [@DamienTan](https://github.com/DamienTan)
> 
> Yep, I used the argument "covar_mod" to add 2 covariates, but it took me more than 4 days and was still running. Finally, I killed this process. Maybe only using the argument "group" can avoid this problem, but there is another problem, which will create some very large values at the adjusted matrix. [#20](https://github.com/zhangyuqing/ComBat-seq/issues/20)

The large values seem to be because of genes that display low counts in the count matrix. Filtering them out before batch correction with ComBat-seq can help with this over inflation of counts.
